libs/dh/wholesale/feature-request-calculation/src/lib/dh-wholesale-request-calculation.ts

```diff
@@ -186,7 +193,7 @@ export class DhWholesaleRequestCalculationComponent {
           this._selectedEicFunction
         );
 
-        const excludeProcessTypes = this.getExcludedProcessTypes(this._selectedEicFunction);
+        const excludeProcessTypes = await this.getExcludedProcessTypes();
```

Comment:
suggestion: I don't like this await here, the subscribe `next` handler does not accept promises like this. Things might get weird when selecting different actors. May I suggest this signal-based approach which also gets rid of that `combineLatest`/`reduce` logic which seems more complicated than it needs to be:

```ts
private _permissionService = inject(PermissionService);

requestAggregatedMeasuredDataView = this.hasPermission('request-aggregated-measured-data:view');
requestWholesaleSettlementView = this.hasPermission('request-wholesale-settlement:view');

excludeCalculationTypes = computed(() => {
  const aggregation = this.requestAggregatedMeasuredDataView() ? [] : aggregationCalculationTypes;
  const wholesale = this.requestWholesaleSettlementView() ? [] : wholesaleCalculationTypes;
  return [...aggregation, ...wholesale];
});

calculationTypeOptions = computed(() =>
  dhEnumToWattDropdownOptions(CalculationType, null, this.excludeCalculationTypes())
);

hasPermission(permission: Permission) {
  return toSignal(this._permissionService.hasPermission(permission));
}
```

Reply:
I like ❤️